server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -680,5 +640,28 @@ long getInitialTermBeforeJoiningALeader() {
         public void close() {
             assertTrue(openPersistedStates.remove(this));
         }
+
+        public ClusterState getLatestStoredStateIfAppliedStateIsStale() {
+            var latestClusterState = sharedStore.getLatestClusterState();
+            if (isLatestAcceptedStateStale(latestClusterState) == false) {
+                return null;
+            }
+
+            return ClusterStateUpdaters.addStateNotRecoveredBlock(
+                ClusterState.builder(new ClusterName("elasticsearch"))
+                    .metadata(latestClusterState.state())
+                    .version(latestClusterState.version())
+                    .blocks(latestAcceptedState.blocks())
+                    .nodes(DiscoveryNodes.builder(latestAcceptedState.nodes()).masterNodeId(null))
+                    .build()
```

Comment:
Hmm this is effectively a full-cluster restart, it wipes out the routing table so we have to go through recovery for every shard. It also drops all `ClusterState.Custom`, and applies a set of blocks that might be inconsistent with the settings in the metadata. I think we discussed this before but I don't remember the conclusion we reached and it seems it isn't mentioned in the design docs.

Is the intention to reset everything for now and optimise things later?

Reply:
The plan is to store that information (or at least enough information to rebuild it) in the blob store. But for now I wanted to reflect what we have. We can discuss this further if you want.

Reply:
Ok, this is fine for now (except we should use `ClusterStateUpdaters.recoverClusterBlocks` to copy the blocks from the metadata)